### PR TITLE
zabbix: Support for 5.4

### PR DESCRIPTION
**Pull request by mc-fugger**
###### Motivation for this change

Zabbix' 5.4 branch wasn't previously supported

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

This adds zabbix54 as extra available packages, zabbix50 stays the default package. I've tested the packages with them being pulled in by NixOS' configuration.nix and so far couldn't find any errors.
I don't really know much about git or nix so I hope everything's alright :sweat_smile:

### CODE REVIEW

**Comment by aanderse** [comment]
I'm sorry but we can't support this software! EOL is Jan 30, 2022, which is during the release cycle of 21.11. See https://www.zabbix.com/life_cycle_and_release_policy for details.

We should add this to NUR instead. If you don't want to do that I could.

**Comment by mc-fugger** [comment]
Ah, so nixpkgs-unstable still follows the stable bi-yearly releases instead of being rolling release on bleeding edge? When 6.0 comes out, it would replace this version as the latest one (like linuxPackages_latest). Would renaming the package to zabbix_latest work?

**Comment by aanderse** [comment]
We don't allow breaking changes during a 6 month release cycle of NixOS - so we can't update `zabbix_latest` to a different major version for the next 6 months. This means we end up with an unsupported version of `zabbix` shortly into our release cycle.

What do you think about adding this to NUR?

File: pkgs/servers/monitoring/zabbix/agent2.nix [hunk 1]
```diff
@@ -12,7 +12,10 @@ import ./versions.nix ({ version, sha256 }:
 
     modRoot = "src/go";
 
-    vendorSha256 = "1417qi061xc4m55z0vz420fr7qpi24kw5yj9wq7iic92smakgkjn";
+    vendorSha256 =
+        if "${lib.versions.majorMinor version}" == "5.0" then "1417qi061xc4m55z0vz420fr7qpi24kw5yj9wq7iic92smakgkjn"
+        else if "${lib.versions.majorMinor version}" == "5.4" then "17hx0lnxvc4vlsdp7h66x95rwwh5skk3kynphm605klfix4hjs2k"
+        else throw "unsupported version ${version} for zabbix-agent2";
```
Comment (SuperSandro2000) [inline pkgs/servers/monitoring/zabbix/agent2.nix:18]: Please passthru the vendor sha from versions instead of doing this.

**Comment by aanderse** [comment]
@mc-fugger this is great work you are doing, but we still need to drop the 5.4 release because it is going EOL in a few months. Can you please respond to my NUR comment?

**Comment by mc-fugger** [comment]
Yes, I was hoping for a different answer but to the NUR it is then :grin: 
Would you like to add it there or should I read up on it? Also, what will happen to the package in the NUR when 6.0 releases, making 5.4 in the NUR obsolete?

**Comment by aanderse** [comment]
@mc-fugger sorry I can't give better news, it is just that we can't remove the 5.4 release a month or two into the NixOS 21.11 release. However we **can** add the 6.0 release to NixOS unstable (and stable) once it is out in a few months.

I don't have any repositories on NUR. I maintain my own 5.4 expression and deploy it myself, so you can either do that or submit it to NUR if you think it is useful to others. Either way it won't be necessary much longer... 6.0 is due out soon.

Your PR makes all of this much easier, so personally I **really** appreciate your work here! Thank you! :sparkle:

**Comment by misuzu** [comment]
6.0 is out - https://www.zabbix.com/whats_new_6_0

**Comment by NickCao** [comment]
5.4 has been EOL for a whole year: https://endoflife.date/zabbix

**Pull request closed without merging**